### PR TITLE
Version 3 Update

### PR DESCRIPTION
**Pull request by antontutoveanu**
Updated:
- Kyber-512 noise parameter eta1 increased from 2 to 3
- Kyber-512 compression/decompression parameter dv from 3 to 4 (changes ciphertext size from 736 to 768 bytes)
- More efficient rejection sampling of public matrix A

### CODE REVIEW

**Comment by antontutoveanu** [comment]
Hmm not sure about the test runs but it seems to work manually running in my terminal

**Comment by antontutoveanu** [comment]
Sorry working on the test runs now

**Comment by antontutoveanu** [comment]
Works as intended

File: byteops.go [hunk 1]
```diff
@@ -13,21 +13,46 @@ func byteopsLoad32(x []byte) uint32 {
...
-	for i := 0; i < paramsN/8; i++ {
-		t = byteopsLoad32(buf[4*i:])
-		d = t & 0x55555555
-		d = d + ((t >> 1) & 0x55555555)
-		for j := 0; j < 8; j++ {
-			a = int16((d >> (4*j + 0)) & 0x3)
-			b = int16((d >> (4*j + paramsETA)) & 0x3)
-			r[8*i+j] = a - b
+
+	if eta == 2 {
```
Comment (nadimkobeissi) [inline byteops.go:33]: Hmm.

**Review by nadimkobeissi — COMMENTED** [review]
There are many problems with this PR. I will suggest fixes.

**Comment by nadimkobeissi** [comment]
@antontutoveanu Thank you so much for this!

Merging and then pushing my own fixes.

**Comment by antontutoveanu** [comment]
Changing eta into eta1 and eta2 was interesting to work into this implementation, I see how you made it more efficient/compact. Be sure to update the about section to mention version 3!

**Pull request merged**